cgp/utils.py

```diff
@@ -11,22 +11,42 @@
 
 
 def __check_cache_consistency(fn: str, func: Callable[..., float]) -> None:
+    """Retrieve an entry from the cache, execute the callable with the
+    cached arguments and check whether the return value matches the
+    cached result.
+
+    WARNING: consistency is only checked when a _finite_ return value
+    can be found in the cache.
+
+    """
+    cached_item: Union[Dict[str, Any], None] = __find_item_with_finite_return_value(fn)
+    if cached_item is None:
+        return
+
+    return_value: float = func(*cached_item["args"], **cached_item["kwargs"])
+
+    if not np.isclose(return_value, cached_item["return_value"]):
+        raise RuntimeError(
+            "inconsistent return values"
+            " -- are different functions using the same cache file?"
+            " please use different cache files for different functions"
+        )
+
+
+def __find_item_with_finite_return_value(fn: str) -> Union[Dict[str, Any], None]:
+    """Find a cache entry which has a finite return value."""
     if os.path.isfile(fn):
         with open(fn, "rb") as f:
-            try:
-                cursor: Dict[str, Any] = pickle.load(f)
-            except EOFError:
-                return  # no entry yet, so not possible to check
-
-            cached_item: Dict[str, Any] = list(cursor.values())[0]
-            return_value: float = func(*cached_item["args"], **cached_item["kwargs"])
-
-            if not np.isclose(return_value, cached_item["return_value"]):
-                raise RuntimeError(
-                    "inconsistent return values"
-                    " -- are different functions using the same cache file?"
-                    " please use different cache files for different functions"
-                )
+            while True:
+                try:
+                    cursor: Dict[str, Any] = pickle.load(f)
+                except EOFError:
+                    return None  # no entry yet, so not possible to check
+
+                cached_item: Dict[str, Any] = list(cursor.values())[0]
+                if np.isfinite(cached_item["return_value"]):
+                    return cached_item
+    return None
```

Comment:
coverage declines because of this line. By itself, that's not a big deal, but do we really need this line? Is it ever executed? 
If this is to catch the case that `os.path.isfile(fn) = False`, then we should make it explicit and put an `else:` statement here. And in this case, we should also have a small test for this.

Reply:
this line should be executed whenever the decorator is called for the first time as the cache file does not exist yet, so i'm surprised that it's never encountered :thinking: in any case, i agree an explicit `else` would make this intention clearer.

Reply:
I wouldn't trust coveralls here a 100 % to be honest. But in any case, why don't you add a simple one-liner test case so got that covered explicitly?

Reply:
indeed, `mkstemp` already creates the file so this line was never hit. added a test now.

Reply:
`black` is complaining in test suite.

Why don't you use pre-commit hooks ...
Why don't you use pre-commit hooks ...
Why don't you use pre-commit hooks ...
...
...

Reply:
:weary: fixed